wakaama.cmake

```diff
@@ -2,6 +2,44 @@ set(WAKAAMA_TOP_LEVEL_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}")
 set(WAKAAMA_EXAMPLE_DIRECTORY "${WAKAAMA_TOP_LEVEL_DIRECTORY}/examples")
 set(WAKAAMA_EXAMPLE_SHARED_DIRECTORY "${WAKAAMA_EXAMPLE_DIRECTORY}/shared")
 
+# Logging
+set(WAKAAMA_LOG_LEVEL
+    LOG_DISABLED
+    CACHE STRING "The lowest log level provided at build time"
+)
+set_property(
+    CACHE WAKAAMA_LOG_LEVEL
+    PROPERTY STRINGS
+             LOG_DISABLED
+             DBG
+             INFO
+             WARN
+             ERR
+             FATAL
+)
+
+option(WAKAAMA_LOG_CUSTOM_HANDLER "Provide a custom handler for logging messages" OFF)
+
+set(WAKAAMA_LOG_MAX_MSG_TXT_SIZE
+    200
+    CACHE STRING "The buffer size for the log message (without additional data)"
+)
+
+# Possibility to disable the examples
+option(WAKAAMA_ENABLE_EXAMPLES "Build all the example applications" ON)
+
+# Set the defines for logging configuration
+function(set_defines target)
+    # Logging
+    target_compile_definitions(${target} PUBLIC LWM2M_LOG_LEVEL=LWM2M_${WAKAAMA_LOG_LEVEL})
+
+    if(WAKAAMA_LOG_CUSTOM_HANDLER)
```

Comment:
Why do you use different prefixes in the names for the cmake variable and the macro?

Reply:
The project name is Wakaama. So the CMake options are prefixed that way. Eventually we should also adjust the prefixes in the code.